(Compile-Time) C error: `os` undeclared

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
### V Doctor:
```sh
OS: linux, Ubuntu 20.04.4 LTS (WSL 2)
Processor: 12 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i5-11400H @ 2.70GHz
CC version: cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

getwd: /home/apocryphon/tests/v
vmodules: /home/apocryphon/.vmodules
vroot: /home/apocryphon/packages/v
vexe: /home/apocryphon/packages/v/v
vexe mtime: 2022-07-26 18:05:29
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 d6de533.c976a69

Git version: git version 2.25.1
Git vroot status: weekly.2022.30-7-gc976a691
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452
```

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
### What did you do?
```v
import os

struct Dummy {}

fn (d Dummy) sample(file_name string) {
    println(file_name)
}

fn main() {
    $for method in Dummy.methods {
        if os.args.len > 1 {
            Dummy{}.$method(os.args)
        }
    }
}
```

### What did you expect to see?
> _No errors._
 
### What did you see instead?
`v -cg bug_comptime2.v`
```
/tmp/v_1000/bug_comptime.9653553717289858583.tmp.c:16989: error: 'os' undeclared
builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```
`v -prod -cg bug_comptime2.v`
```
/tmp/v_1000/bug_comptime.17375409208485282420.tmp.c: In function ‘main__main’:
/tmp/v_1000/bug_comptime.17375409208485282420.tmp.c:16963:80: error: ‘os’ undeclared (first use in this function); did you mean ‘tos’?
16963 |     main__Dummy_sample(((main__Dummy){EMPTY_STRUCT_INITIALIZATION}), ((string*)os.args.data) [0] );
      |                                                                                ^~
      |                                                                                tos
/tmp/v_1000/bug_comptime.17375409208485282420.tmp.c:16963:80: note: each undeclared identifier is reported only once for each function it appears in
builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/gen/c/comptime.v
+++ b/vlib/v/gen/c/comptime.v
@@ -147,12 +147,13 @@ fn (mut g Gen) comptime_call(mut node ast.ComptimeCall) {
 			} else {
 				// last argument; try to expand if it's []string
 				idx := i - node.args.len
+				last_arg := g.expr_string(node.args.last().expr)
 				if m.params[i].typ.is_int() || m.params[i].typ.idx() == ast.bool_type_idx {
 					// Gets the type name and cast the string to the type with the string_<type> function
 					type_name := g.table.type_symbols[int(m.params[i].typ)].str()
-					g.write('string_${type_name}(((string*)${node.args[node.args.len - 1]}.data) [$idx])')
+					g.write('string_${type_name}(((string*)${last_arg}.data) [$idx])')
 				} else {
-					g.write('((string*)${node.args[node.args.len - 1]}.data) [$idx] ')
+					g.write('((string*)${last_arg}.data) [$idx] ')
 				}
 				if i < m.params.len - 1 {
 					g.write(', ')

--- a/vlib/v/tests/comptime_for_method_call_with_args_test.v
+++ b/vlib/v/tests/comptime_for_method_call_with_args_test.v
@@ -1,0 +1,17 @@
+import os
+
+struct Dummy {}
+
+fn (d Dummy) sample(file_name string) {
+	println(file_name)
+}
+
+fn test_comptime_for_method_call_with_args() {
+	$for method in Dummy.methods {
+		if os.args.len > 1 {
+			d := Dummy{}
+			d.$method(os.args)
+		}
+	}
+	assert true
+}

